### PR TITLE
Live Edit Validations

### PR DESCRIPTION
**Pull request by Hegazy360**

### CODE REVIEW

File: addon/mixins/editable.js [hunk 1]
```diff
@@ -5,21 +5,30 @@ import { run } from '@ember/runloop';
...
   classNameBindings: [`editStatus.status`],
 
-  editStatus: computed('item.editStatus', 'column.key', function() {
-    if(this.column.key === this.get('item.editStatus.key')) {
-      return this.item.editStatus;
-    }
+  editStatus: computed('manager.editStatus', 'column.key', function() {
+    if (
+      this.item === this.manager.get("editStatus.item") &&
+      this.column.key === this.manager.get("editStatus.key")
```
Comment (phndiaye) [inline addon/mixins/editable.js]: single quotes :p

File: addon/mixins/editable.js [hunk 2]
```diff
@@ -5,21 +5,30 @@ import { run } from '@ember/runloop';
...
+        if(this.value !== this.editableValue) {
+          this.set('editableValue', this.value)
+        }
+        this.manager.set('editStatus', {key: this.column.key, status: 'editing', item: this.item})
         run.scheduleOnce('afterRender', this, () => {
           this.$('.editing-input__field').focus();
         });
       } else if(this.get('editStatus.status') !== 'success') {
-        this.manager.hooks.onLiveEdit({key: this.column.key, field: this.item, value});
+        this.manager.hooks.onLiveEdit({key: this.column.key, field: this.item, value})
```
Comment (phndiaye) [inline addon/mixins/editable.js]: semicolon at the end

File: addon/components/hyper-table/cell.js [hunk 3]
```diff
@@ -15,7 +13,7 @@ export default Component.extend({
   renderingComponent: null,
 
   click(e) {
-    if (this.manager.hooks.onRowClicked && !this.item.editStatus) {
+    if (this.manager.hooks.onRowClicked && (!this.manager.editStatus) || this.manager.get('editStatus.status')) {
```
Comment (tgallice) [inline addon/components/hyper-table/cell.js]: parenthesis around `!this.manager.editStatus` are not needed

File: addon/mixins/editable.js [hunk 4]
```diff
@@ -5,21 +5,40 @@ import { run } from '@ember/runloop';
...
   classNameBindings: [`editStatus.status`],
 
-  editStatus: computed('item.editStatus', 'column.key', function() {
-    if(this.column.key === this.get('item.editStatus.key')) {
-      return this.item.editStatus;
-    }
+  editStatus: computed('manager.editStatus', 'column.key', function() {
+    if (
+      this.item === this.manager.get('editStatus.item') &&
+      this.column.key === this.manager.get('editStatus.key')
```
Comment (tgallice) [inline addon/mixins/editable.js]:
```js
let { item, key } = this.manager.getWithDefault('editStatus', {});
if (this.item = item && this.column.key === key) {
....
}
```

File: addon/mixins/editable.js [hunk 5]
```diff
@@ -5,21 +5,40 @@ import { run } from '@ember/runloop';
...
+  isSuccess: computed.equal('editStatus.status', 'success'),
+  isEditing: computed.equal('editStatus.status', 'editing'),
+  isSaving: computed.equal('editStatus.status', 'saving'),
+  isError: computed.equal('editStatus.status', 'error'),
+
   actions: {
     toggleEditing(value) {
-      if(!this.get('editStatus.status')){
-        this.item.set('editStatus', {key: this.column.key, status: 'editing'})
+      if (!this.get('editStatus.status') || this.manager.get('editStatus.status') === 'success') {
```
Comment (tgallice) [inline addon/mixins/editable.js]:
```js
let status = this.manager.getWithDefault('editStatus.status', 'success');

if (status !== 'success') {
        this.manager.hooks.onLiveEdit({
          key: this.column.key,
          field: this.item,
          value
        });
    return
} 

...

```

File: addon/mixins/editable.js [hunk 6]
```diff
@@ -5,22 +5,50 @@ import { run } from '@ember/runloop';
 export default Mixin.create({
   classNameBindings: [`editStatus.status`],
 
-  editStatus: computed('item.editStatus', 'column.key', function() {
-    if(this.column.key === this.get('item.editStatus.key')) {
-      return this.item.editStatus;
-    }
+  editStatus: computed('manager.editStatus', 'column.key', function() {
+    let { item, key } = this.manager.editStatus || {}
+    if (this.item === item && this.column.key === key)
```
Comment (tgallice) [inline addon/mixins/editable.js]: please use `{}` :)

File: app/templates/components/hyper-table/cell-renderers/text.hbs [hunk 7]
```diff
@@ -1,20 +1,21 @@
-{{#if (and emptyValue (or (not editStatus.status) (eq editStatus.status "success")))}}
+{{#if (and emptyValue (or (not editStatus.status) isSuccess))}}
   {{#if column.upsertable}}
-    Add Value
+    <span onclick={{action "toggleEditing"}}>
```
Comment (phndiaye) [inline app/templates/components/hyper-table/cell-renderers/text.hbs]: `{{action x}}` = `onclick={{action x}}`

**Pull request merged**